src/main/scala/shine/C/CodeGeneration/CodeGenerator.scala

```diff
@@ -496,8 +496,17 @@ class CodeGenerator(val decls: CodeGenerator.Declarations,
       case b: shine.DPIA.Types.BasicType => b match {
         case shine.DPIA.Types.bool => C.AST.Type.int
         case shine.DPIA.Types.int | shine.DPIA.Types.NatType => C.AST.Type.int
-        case shine.DPIA.Types.float => C.AST.Type.float
-        case shine.DPIA.Types.double => C.AST.Type.double
+        case shine.DPIA.Types.u8 => C.AST.Type.u8
+        case shine.DPIA.Types.u16 => C.AST.Type.u16
+        case shine.DPIA.Types.u32 => C.AST.Type.u32
+        case shine.DPIA.Types.u64 => C.AST.Type.u64
+        case shine.DPIA.Types.i8 => C.AST.Type.i8
+        case shine.DPIA.Types.i16 => C.AST.Type.i16
+        case shine.DPIA.Types.i32 => C.AST.Type.i32
+        case shine.DPIA.Types.i64 => C.AST.Type.i64
+        case shine.DPIA.Types.f16 => ???
```

Comment:
Do we want this if we cannot use it? We can reintroduce it when we introduce Tensor Cores or something else that requires half-precision floating point values.

Reply:
We can use it with the OpenCL extension: https://www.khronos.org/registry/OpenCL/sdk/1.0/docs/man/xhtml/cl_khr_fp16.html
There must also be a way to use this in plain C.
In general, we should be able to use it but it requires a bit more thought.
I think that having the type around with most of the code paths is good.

Reply:
There definitely is a way in C as well. It sounds to me that you want to keep it as a reminder that there can be half-precision as well? Or do you have a specific use case in mind?

Reply:
There are many use cases in image processing and I have used half precision before during my internships. I have no plan to use them short term, but in the long term we will want them.